pkg/services/ngalert/api/forking_alertmanager.go

```diff
@@ -1,36 +1,83 @@
 package api
 
 import (
+	"net/http"
+	"strings"
+
+	amv2 "github.com/prometheus/alertmanager/api/v2/models"
+
 	"github.com/grafana/grafana/pkg/api/response"
 	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
 	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/web"
 )
 
+const extraConfigPrefix = "__grafana-converted-extra-config-"
```

Comment:
WDYT about making the path segment an invalid UID, i.e. that does not match https://github.com/grafana/grafana/blob/44fd13c742e606b8409e23eb62cab8bab24310f1/pkg/util/shortid_generator.go#L33 . This will greatly reduce possibility of collision with an existing data source.

Reply:
Good idea 👍